shared/src/commonMain/kotlin/com/devil/phoenixproject/ui/sync/LinkAccountViewModel.kt

```
@@ -83,6 +87,70 @@ class LinkAccountViewModel(private val syncManager: SyncManager) {
+        scope.launch {
+            try {
+                _uiState.value = LinkAccountUiState.Loading
+                authRepository.signInWithGoogle()
+                    .onSuccess { finishOAuthLink() }
+                    .onFailure { error ->
+                        _uiState.value = LinkAccountUiState.Error(
+                            error.message ?: "Google sign-in failed",
+                        )
+                    }
+            } catch (e: CancellationException) {
+                throw e
```

detekt (1.23.8)

[warning] Do not rethrow a caught exception of the same type.

(detekt.exceptions.RethrowCaughtException)